src/plugins/data/public/search/aggs/agg_types.ts

```diff
@@ -52,44 +50,43 @@ import { bucketAvgMetricAgg } from './metrics/bucket_avg';
 import { bucketMinMetricAgg } from './metrics/bucket_min';
 import { bucketMaxMetricAgg } from './metrics/bucket_max';
 
-export function getAggTypes(deps: { uiSettings: IUiSettingsClient }) {
-  const { uiSettings } = deps;
-  return {
-    metrics: [
-      countMetricAgg,
-      avgMetricAgg,
-      sumMetricAgg,
-      medianMetricAgg,
-      minMetricAgg,
-      maxMetricAgg,
-      stdDeviationMetricAgg,
-      cardinalityMetricAgg,
-      percentilesMetricAgg,
-      percentileRanksMetricAgg,
-      topHitMetricAgg,
-      derivativeMetricAgg,
-      cumulativeSumMetricAgg,
-      movingAvgMetricAgg,
-      serialDiffMetricAgg,
-      bucketAvgMetricAgg,
-      bucketSumMetricAgg,
-      bucketMinMetricAgg,
-      bucketMaxMetricAgg,
-      geoBoundsMetricAgg,
-      geoCentroidMetricAgg,
-    ],
-    buckets: [
-      dateHistogramBucketAgg,
-      histogramBucketAgg,
-      rangeBucketAgg,
-      dateRangeBucketAgg,
-      ipRangeBucketAgg,
-      termsBucketAgg,
-      filterBucketAgg,
-      getFiltersBucketAgg({ uiSettings }),
-      significantTermsBucketAgg,
-      geoHashBucketAgg,
-      geoTileBucketAgg,
-    ],
-  };
-}
+import { AggTypesDependencies } from './types';
+
+export const getAggTypes = (deps: AggTypesDependencies) => ({
+  metrics: [
+    countMetricAgg,
+    avgMetricAgg,
+    sumMetricAgg,
+    medianMetricAgg,
+    minMetricAgg,
+    maxMetricAgg,
+    stdDeviationMetricAgg,
+    cardinalityMetricAgg,
+    percentilesMetricAgg,
+    percentileRanksMetricAgg,
+    topHitMetricAgg,
+    derivativeMetricAgg,
+    cumulativeSumMetricAgg,
+    movingAvgMetricAgg,
+    serialDiffMetricAgg,
+    bucketAvgMetricAgg,
+    bucketSumMetricAgg,
+    bucketMinMetricAgg,
+    bucketMaxMetricAgg,
+    geoBoundsMetricAgg,
+    geoCentroidMetricAgg,
+  ],
+  buckets: [
+    getDateHistogramBucketAgg(deps),
+    getHistogramBucketAgg(deps),
+    rangeBucketAgg,
+    getDateRangeBucketAgg(deps),
+    ipRangeBucketAgg,
+    termsBucketAgg,
+    filterBucketAgg,
+    getFiltersBucketAgg(deps),
```

Comment:
I think we should pass _only the required dependencies_ to each agg type, rather than `AggTypesDependencies`, which includes all of `core` and all of the `query` service. (Similar to how `getFiltersBucketAgg` was previously implemented)

Each of these agg types only uses a tiny fraction of each of these services -- for the sake of maintainability I would much rather be explicit about which service dependencies an agg type actually has... that way we don't have to go opening up every single agg type file in the future if we want to find out what the dependencies are.

Another reason for doing this is that we are eventually going to need to support aggs on the server, which makes it even more important that we aren't coupling these to the public `CoreSetup`, as this is already going to be different from the server `CoreSetup`.

So my suggestion would be:
- Only pass the required services into `getAggTypes`
- Only pass the required services into each individual agg (this should be a subset of what's provided to `getAggTypes`